packages/core/src/RenderingEngine/StackViewport.ts

```diff
@@ -1573,32 +1579,44 @@ class StackViewport extends Viewport implements IStackViewport {
 
     // 1. Update the pixel data in the vtkImageData object with the pixelData
     //    from the loaded Cornerstone image
-    const pixelData = image.getPixelData();
+    let pixelData = image.getPixelData();
     const scalars = this._imageData.getPointData().getScalars();
     const scalarData = scalars.getData() as
       | Uint8Array
       | Float32Array
       | Uint16Array
       | Int16Array;
 
-    if (image.rgba || isRgbaSourceRgbDest(pixelData, scalarData)) {
-      if (!image.rgba) {
-        console.warn('rgba not specified but data looks rgba ish', image);
-      }
-      // if image is already cached with rgba for any reason (cpu fallback),
-      // we need to convert it to rgb for the pixel data set
-      // RGB case
-      const numPixels = pixelData.length / 4;
-
-      let rgbIndex = 0;
-      let index = 0;
-
-      for (let i = 0; i < numPixels; i++) {
-        scalarData[index++] = pixelData[rgbIndex++]; // red
-        scalarData[index++] = pixelData[rgbIndex++]; // green
-        scalarData[index++] = pixelData[rgbIndex++]; // blue
-        rgbIndex++; // skip alpha
+    // if the color image is loaded with CPU previously, it loads it
+    // with RGBA, and here we need to remove the A channel from the
+    // pixel data.
+    if (image.color && image.rgba) {
+      const newPixelData = new Uint8Array(image.columns * image.rows * 3);
+      for (let i = 0; i < image.columns * image.rows; i++) {
+        newPixelData[i * 3] = pixelData[i * 4];
+        newPixelData[i * 3 + 1] = pixelData[i * 4 + 1];
+        newPixelData[i * 3 + 2] = pixelData[i * 4 + 2];
       }
+      pixelData = newPixelData;
+
+      //  const numPixels = pixelData.length / 4;
+      //  const pixelDataWithoutAlpha = new Uint8Array(numPixels * 3);
+
+      //  let rgbaIndex = 0;
+      //  let rgbIndex = 0;
+
+      //  for (let i = 0; i < numPixels; i++) {
+      //    pixelDataWithoutAlpha[rgbIndex++] = pixelData[rgbaIndex++]; // red
+      //    pixelDataWithoutAlpha[rgbIndex++] = pixelData[rgbaIndex++]; // green
+      //    pixelDataWithoutAlpha[rgbIndex++] = pixelData[rgbaIndex++]; // blue
+      //    rgbaIndex++; // skip alpha
+      //  }
+
+      // modify the image object to have the correct pixel data for later
+      // use.
+      image.rgba = false;
```

Comment:
Any way the image could be replaced rather than being modified in place?  Will this ever cause issues with other display environments needing rgba such as the thumbnails?

Reply:
This is an edge case + optimization where image is rendered with rgba first, I guess I can make it not replace at all just uses the new pixelData, but every other call will do the same operation. Plus, I have a PR to make thumbnails use GPU soon